src/main/java/at/hannibal2/skyhanni/features/misc/ChestValue.kt

```diff
@@ -0,0 +1,114 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.DrawScreenAfterEvent
+import at.hannibal2.skyhanni.events.InventoryOpenEvent
+import at.hannibal2.skyhanni.features.misc.items.EstimatedItemValue
+import at.hannibal2.skyhanni.utils.GuiRenderUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils.addOrPut
+import at.hannibal2.skyhanni.utils.LorenzUtils.round
+import io.github.moulberry.notenoughupdates.mixins.AccessorGuiContainer
+import io.github.moulberry.notenoughupdates.util.GuiTextures.accessory_bag_overlay
+import io.github.moulberry.notenoughupdates.util.Utils
+import net.minecraft.client.Minecraft
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraft.item.ItemStack
+import net.minecraft.util.EnumChatFormatting
+import net.minecraftforge.fml.common.eventhandler.EventPriority
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import org.lwjgl.opengl.GL11
+import java.awt.Color
+import java.util.*
+
+
+class ChestValue {
+
+    private val config get() = SkyHanniMod.feature.misc
+
+    var data = mutableMapOf<Int, MutableList<ItemStack>>()
+
+    private var itemValueTotal = 0.0
+    private var itemEntryList: MutableList<Map.Entry<ItemStack, Double>> = mutableListOf()
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    fun onInventoryOpen(event: InventoryOpenEvent) {
+        if (!LorenzUtils.inSkyBlock) return
+        if (!config.estimatedChestValue) return
+
+        var value = 0.0
+        var valueMap: MutableMap<ItemStack, Double> = mutableMapOf()
+
+        for ((slot, item) in event.inventoryItems) {
+            value += EstimatedItemValue.getEstimatedItemPrice(item, mutableListOf()).first * item.stackSize
+            valueMap.addOrPut(item, EstimatedItemValue.getEstimatedItemPrice(item, mutableListOf()).first * item.stackSize)
+        }
+        itemValueTotal = value
+        var i = 0
+
+        itemEntryList.clear()
+        while (i<=3){
```

Comment:
In IntelliJ, you can use the shortcut `control + alt + L` to apply the auto formatting tool to the whole file